tests/BreadcrumbsTest.php

```diff
@@ -4,197 +4,416 @@
+    {
+        Assert::equalsWithoutLE(
+            <<<HTML
+            <nav aria-label="Basic example of breadcrumbs">
+            <ol class="breadcrumb">
+            <li class="breadcrumb-item"><a href="/">Home</a></li>
+            <li class="breadcrumb-item"><a href="#">Library</a></li>
+            <li class="breadcrumb-item active" aria-current="page">Data</li>
+            </ol>
+            </nav>
+            HTML,
+            Breadcrumbs::widget()
+                ->ariaLabel('Basic example of breadcrumbs')
+                ->links(
+                    new BreadcrumbLink('Home', '/'),
+                    new BreadcrumbLink('Library', '#'),
+                    new BreadcrumbLink('Data'),
+                )
+                ->listId(false)
+                ->render(),
+        );
+    }
+
+    public function testAttributes(): void
     {
-        $html = Breadcrumbs::widget()
-            ->id('test')
-            ->homeLink(['label' => 'Home', 'url' => '#'])
-            ->links([
-                ['label' => 'Library', 'url' => '#', 'template' => "<span class=\"testMe\">{link}</span>\n"],
-                ['label' => 'Data'],
-                'Articles',
-            ])
-            ->render();
-        $expected = <<<'HTML'
-        <nav aria-label="breadcrumb"><ol id="test-breadcrumb" class="breadcrumb"><li class="breadcrumb-item"><a href="#">Home</a></li>
-        <span class="testMe"><a href="#">Library</a></span>
-        <li class="breadcrumb-item active" aria-current="page">Data</li>
-        <li class="breadcrumb-item active" aria-current="page">Articles</li>
-        </ol></nav>
-        HTML;
-        $this->assertSame($expected, $html);
+        Assert::equalsWithoutLE(
+            <<<HTML
+            <nav class="test-class" aria-label="breadcrumb">
+            <ol class="breadcrumb">
+            <li class="breadcrumb-item"><a href="/">Home</a></li>
+            <li class="breadcrumb-item"><a href="#">Library</a></li>
+            <li class="breadcrumb-item active" aria-current="page">Data</li>
+            </ol>
+            </nav>
+            HTML,
+            Breadcrumbs::widget()
+                ->attributes(['class' => 'test-class'])
+                ->links(
+                    new BreadcrumbLink('Home', '/'),
+                    new BreadcrumbLink('Library', '#'),
+                    new BreadcrumbLink('Data'),
+                )
+                ->listId(false)
+                ->render(),
+        );
     }
 
-    public function testMissingLinks(): void
+    /**
+     * @link https://getbootstrap.com/docs/5.2/components/breadcrumb/#dividers
+     */
+    public function testDivider(): void
     {
-        $html = Breadcrumbs::widget()
-            ->links([])
-            ->render();
-        $this->assertEmpty($html);
+        Assert::equalsWithoutLE(
+            <<<HTML
+            <nav style="--bs-breadcrumb-divider: &apos;&gt;&apos;;" aria-label="breadcrumb">
```

Comment:
How does it work with totally custom style element? Do browsers support it?

Reply:
![image](https://github.com/user-attachments/assets/9f8623f0-f17e-4aca-a30f-5eed66489384)